chapters/operatorsandexpressions.tex

```diff
@@ -1644,10 +1644,14 @@ \subsection{Discrete-Time Expressions}\label{discrete-time-expressions}
   Note that \lstinline!rem! and \lstinline!mod! generate events but are not discrete-time expressions.
   In other words, relations inside \lstinline!noEvent!, such as \lstinline!noEvent(x>1)!, are not discrete-time expressions.
   \end{nonnormative}
+\item
+  Unless inside \lstinline!noEvent!: Function calls where the function has annotation \lstinline!GenerateEvents = true! (\Cref{modelica:GenerateEvents}) and the output does not contain a subtype of \lstinline!Real!.
+  For a function call returning multiple return values (see \cref{output-formal-parameters-of-functions}) the variability is decided separately for each output.
+  These calls will generate events, \Cref{modelica:GenerateEvents}.
 \item
   The functions \lstinline!pre!, \lstinline!edge!, and \lstinline!change! result in discrete-time expressions.
 \item
-  Expressions in functions behave as though they were discrete-time expressions.
+  Expressions in functions not having annotation \lstinline!GenerateEvents = true! (\Cref{modelica:GenerateEvents}), behave as though they were discrete-time expressions.
```

Comment:
Don't we need more specification of variability inside functions that do have `GenerateEvents = true`?  We don't want all expressions inside such function bodies to get non-discrete-time variability, right?

Reply:
This could be a start…
```suggestion
  The variability of expressions inside functions depends upon the \lstinline!GenerateEvents! annotation (\Cref{modelica:GenerateEvents}).  With \lstinline!GenerateEvents = true!, expression variability is determined according to the same rules as outside of functions.  Otherwise, all expressions have at most discrete-time variability (some expressions, such as function inputs and literals have lower variability).```

Reply:
That was the intent of the text I wrote (ignoring the misplaced trailing quotes).

The overall structure of that secition is that it specifies when expressions have lower variability. Stating that functions having GenerateEvents=true use the same rule as non-function is thus not needed, and similarly there is no need to say "at most discrete-time" as the less variable cases should already be handled.

Reply:
OK, makes sense.  The formulation is a bit unusual, though, shouldn't we simply say:
```suggestion
  Expressions in functions not having annotation \lstinline!GenerateEvents = true! (\Cref{modelica:GenerateEvents}).
```
This item would then look better before or after _Expressions in a clocked discrete-time partition_.

I also think there is a missing `\cref{discrete-time-expressions}` link to this section from this item in _functions.tex_, and the item would need to be updated according to whatever semantics we come up with here:
> Components of a function will inside the function behave as though they had discrete-time variability.

Reply:
The "behave as though they were discrete-time expressions." have been present for quite some time on some of the entries. I don't want to change them as part of this PR.

Reply:
And added the other reference.